Add traceback to VmException (#657)
lambdaclass/cairo-rs: CHANGELOG.md

2022-12-28

> * Add traceback to VmException
>     * PR: [#657](https://github.com/lambdaclass/cairo-rs/pull/657)
>     * Main functionality changes: `VmException` now contains a traceback in the form of a String which lists the locations (consisting of filename, line, column and pc) of the calls leading to the error.

Reviewer: This can be more concise:
```suggestion
* Add traceback to VmException [#657](https://github.com/lambdaclass/cairo-rs/pull/657)
```

Author: Sounds good!

    * Public API changes: 

>         * `traceback` field added to `VmException` struct
>         * `VmException::from_vm_error()` signature changed from `pub fn from_vm_error(runner: &CairoRunner, error: VirtualMachineError, pc: usize) -> Self` to `pub fn from_vm_error(runner: &CairoRunner, vm: &VirtualMachine, error: VirtualMachineError) -> Self`
>         * `get_location()` signature changed from `pub fn get_location(pc: &usize, runner: &CairoRunner) -> Option<Location>` to `pub fn get_location(pc: usize, runner: &CairoRunner) -> Option<Location>`
>         * `decode_instruction()` signature changed from `pub fn decode_instruction(encoded_instr: i64, mut imm: Option<BigInt>) -> Result<instruction::Instruction, VirtualMachineError>` to `pub fn decode_instruction(encoded_instr: i64, mut imm: Option<&BigInt>) -> Result<instruction::Instruction, VirtualMachineError>`
>         * Minor changes in `VmExcepion` field's string format to mirror their cairo-lang conterparts.

Reviewer: These may only mention the before and after, like:
```suggestion
        * `traceback` field added to `VmException` struct
        * `pub fn from_vm_error(runner: &CairoRunner, error: VirtualMachineError, pc: usize) -> Self` is now `pub fn from_vm_error(runner: &CairoRunner, vm: &VirtualMachine, error: VirtualMachineError) -> Self`
        * `pub fn get_location(pc: &usize, runner: &CairoRunner) -> Option<Location>` is now `pub fn get_location(pc: usize, runner: &CairoRunner) -> Option<Location>`
        * `pub fn decode_instruction(encoded_instr: i64, mut imm: Option<BigInt>) -> Result<instruction::Instruction, VirtualMachineError>` is now `pub fn decode_instruction(encoded_instr: i64, mut imm: Option<&BigInt>) -> Result<instruction::Instruction, VirtualMachineError>`
        * `VmExcepion` field's string format now mirror their cairo-lang conterparts.
```

In any case, I think it's better to signal a given release breaks the API instead and keep a migration guide for the details. We also don't need dates IMO. Users are expected to use releases and the changelogs are only for them, so that's how we should group the changes. Besides, the information given by pointing to the PR is a superset of the one given by the date, making it redundant.

Author: Done!
